### PR TITLE
EN-949: fix variable validations (patch)

### PR DESCRIPTION
**Pull request by chrisjaimon2012**

### CODE REVIEW

File: modules/azure-virtual-network-gateway/vars.tf [hunk 1]
```diff
@@ -17,14 +17,6 @@ variable "ip_configuration" {
...
     error_message = "Valid options for `ip_configuration.private_ip_address_allocation` are \"Static\" or \"Dynamic\"."
   }
-  validation {
-    condition = alltrue([
-      # TODO: change to endswith()
-      # https://github.com/hashicorp/terraform/pull/31220
-      for ip_config in var.ip_configuration : can(regex("GatewaySubnet$", ip_config["subnet_id"]))
-    ])
-    error_message = "It is mandatory that the associated subnet is named \"GatewaySubnet\"."
-  }
```
Comment (GerardSetho) [inline modules/azure-virtual-network-gateway/vars.tf:27]: So it's no longer a requirement that `subnet_id` must end with `GatewaySubnet`?

Comment (chrisjaimon2012) [reply modules/azure-virtual-network-gateway/vars.tf:27]: it is, but we can't validate this, as `ID` is only generated `post-apply`

**Review by GerardSetho — APPROVED** [review]

**Comment by chrisjaimon2012** [comment]
Turns out, this doesn't work. Creating a new PR to fix the syntax

**Comment by chrisjaimon2012** [comment]
https://github.com/quantum-sec/package-azure-networking/pull/18

**Pull request merged**